independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/multipart/FileUpload.java

```diff
@@ -0,0 +1,44 @@
+package org.jboss.resteasy.reactive.multipart;
+
+import java.nio.file.Path;
+
+/**
+ * Represents a file-upload from an HTTP multipart form submission.
+ */
+public interface FileUpload {
+
+    /**
+     * @return the name of the upload as provided in the form submission.
+     */
+    String name();
+
+    /**
+     * @return the actual temporary file name on the server where the file was uploaded to.
+     */
+    Path uploadedFile();
```

Comment:
This assumes everything is a file, which is often not the case. If there is no associated file name then this is just a normal file upload (e.g. a text box). IMHO we should only be creating files for the actual file parts of the request, and provide a convenient way to get the non-file parts.

Reply:
In light of my comment below, do you think this still needs to be addressed?

Reply:
Ah, in that case I think it is fine.